### ISSUE TITLE
incorrect formatting on optional function parameters when parameters on separate lines

### ISSUE THREAD

**Issue by DetachHead** [dart-lang/dart_style#1134]
# before
```dart
void foo(final String value1, final String value2, { final String value3 = 'asdf',}) {}
```
# after
## expected
```dart
void foo(
    final String value1,
    final String value2, {
      final String value3 = 'asdf',
    }
) {}
```
## actual
```dart
void foo(
  final String value1,
  final String value2, {
  final String value3 = 'asdf',
}) {}
```

**Comment by srawlins** [comment]
Why do you expect named or positional-optional parameters to be indented differently from required positional parameters?

**Comment by srawlins** [comment]
Sorry to belabor the point, but could you expand on the reason? The braces that wrap positional-optional parameters and named parameters do not create a new scope separate from required parameters. The braces do not introduce a statement block, as in `if (condition {}`. Is it just the presence of the braces?

**Comment by DetachHead** [comment]
i think it's more about consistency/readability than scope. when i see a `{`, i expect the content within it to be indented once and the `}` to be unindented back to the same level as the line with the `{`.

i guess a more similar example would be something like this:
```dart
final foo = {
  'a': 1,
  'b': {
  'c': 2
}};
```

what about if it was more like this?
```dart
void foo(
    final String value1,
    final String value2,
    {
      final String value3 = 'asdf',
    }
) {}
```

**Comment by munificent** [comment]
This formatting is deliberate. I don't love it either, but it seems to be the least bad option from the several ones I tried. Most users seem happy with it as far as I can tell. And, when I first implemented this formatting behavior, it seemed to be the style that people hand-formatting their code used most often. So, while it's not ideal, I think the current output is probably what we'll stick with.
